packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -178,6 +196,119 @@ export class PaginatedMessage {
+	 * const { MessageBuilder, PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const embed = new MessageEmbed()
+	 * 	.setColor('#FF0000')
+	 * 	.setDescription('example description');
+	 *
+	 * const builder = new MessageBuilder()
+	 * 	.setContent('example content')
+	 * 	.setEmbed(embed);
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageBuilder(builder);
+	 * ```
+	 */
+	public addPageBuilder(builder: MessageBuilder | ((builder: MessageBuilder) => MessageBuilder)): this {
+		return this.addPage(isFunction(builder) ? builder(new MessageBuilder()) : builder);
+	}
+
+	/**
+	 * Adds a page to the existing ones using simple message content. This will be added as the last page.
+	 * @param content The content to set.
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageContent('example content');
+	 * ```
+	 */
+	public addPageContent(content: string): this {
+		return this.addPage({ content });
+	}
+
+	/**
+	 * Adds a page to the existing ones using a {@link MessageEmbed}. This wil be added as the last page.
+	 * @param embed Either a callback whose first paramter is `new MessageEmbed()`, or an already constructed {@link MessageEmbed}
+	 * @example
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageEmbed((embed) => {
+	 * 		embed
+	 * 			.setColor('#FF0000')
+	 * 			.setDescription('example description');
+	 *
+	 * 		return embed;
+	 * });
+	 * ```
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const embed = new MessageEmbed()
+	 * 	.setColor('#FF0000')
+	 * 	.setDescription('example description');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageEmbed(embed);
+	 * ```
+	 */
+	public addPageEmbed(embed: MessageEmbed | ((embed: MessageEmbed) => MessageEmbed)): this {
+		return this.addPage({ embed: isFunction(embed) ? embed(new MessageEmbed()) : embed });
+	}
+
+	/**
+	 * Adds a page to the existing ones asynchronously using a {@link MessageEmbed}. This wil be added as the last page.
+	 * @param embed Either a callback whose first paramter is `new MessageEmbed()`, or an already constructed {@link MessageEmbed}
+	 * @example
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageEmbed(async (embed) => {
+	 *
+	 *		const someRemoteData = await fetch('https://contoso.com/api/users');
+	 *
+	 * 		embed
+	 * 			.setColor('#FF0000')
+	 * 			.setDescription(someRemoteData.data);
+	 *
+	 * 		return embed;
+	 * });
+	 * ```
+	 */
+	public addAsyncPageEmbed(cb: MessageEmbed | ((builder: MessageEmbed) => Promise<MessageEmbed>)): this {
```

Comment:
Why just embed? Why not content too, or hell, anything else

Reply:
That's what `addPageBuilder` and `addPageContent` are for. The former doing everything from `MessageOptions`.

But for the actual reason, I only needed this in Skyra and this is a backport after all. I'll add `addAsyncPageBuilder`. There is no point in `addAsyncPageContent` because `addPageContent` doesn't take a callback.